frontend/src/components/Room.tsx

```diff
@@ -163,11 +171,27 @@ class Room extends React.Component<Props, State> {
     this.setState({ videoQueue: videoQueue.filter(video => video.id !== id) });
   }
 
+  setUsernameAndEmit(): void {
+    if (typeof this.props.location.state === "undefined" || this.props.location.state.username === "") {
+      const randomName: string = uniqueNamesGenerator(customNameConfig);
+      this.setState(
+        {
+          userName: randomName
+        },
+        () => {
+          this.socket.emit(Event.CREATE_USERNAME, this.state.userName);
+        }
+      );
+    } else {
+      this.socket.emit(Event.CREATE_USERNAME, this.state.userName);
```

Comment:
I think `username` is one word so there should be no capital `N` in the middle

Reply:
Done! :D